pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -390,10 +419,33 @@ func addColumn(b BuildCtx, spec addColumnSpec) (backing *scpb.PrimaryIndex) {
 		updatedPartitioning.IndexID = temp.IndexID
 		b.Add(updatedPartitioning)
 	}
-
+	for _, ec := range append(existingColumns, ic) {
+		cloned := protoutil.Clone(ec).(*scpb.IndexColumn)
+		cloned.IndexID = temp.IndexID
+		b.Add(cloned)
+	}
```

Comment:
Wow... Now we would have 3 sets of index columns for the old primary index, new primary index, and temp index; Or 5 sets if this newly added column is UNIQUE and hence another secondary index (and its temp index) needs to be created.

 The stage graph is going to get BIG!